### ISSUE TITLE
Accept header parsing possibly too strict

### ISSUE THREAD

**Issue by dannycoates** [sveltejs/kit#4962]
### Describe the bug

Hi! I noticed some 500 responses in our server logs and tracked it down to this line in SvelteKit:

https://github.com/sveltejs/kit/blob/27a9888161b86bcc8418f87f93ed54959c631079/packages/kit/src/runtime/server/page/index.js#L57

The Accept header in question was `text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2`, with the part throwing the error being `*; q=.2`. 

The user agent was `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36`

### Reproduction

Here's a test:

https://github.com/dannycoates/kit/blob/33e1215a10a172f205114228aa7da829a85e11f2/packages/kit/test/apps/basics/test/test.js#L625-L636

https://github.com/dannycoates/kit/commit/33e1215a10a172f205114228aa7da829a85e11f2

IMO it should return a 200 with the `text/html` type.

### System Info

```shell
N/A
```

### Severity

annoyance

### Additional Information

More than an annoyance tbh. The user sees a 500 error page. We just haven't seen too many of these in our logs yet.

**Comment by lundsholm** [comment]
@dannycoates, I was troubleshooting this and read through the MDN docs about the accept header:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept

From their examples it looks like the asterisk "*" can be used as either the second, or both parts of the MIME type value.
That would make the part you've hightlighted (```*;q=.2```) an invalid header value.

```*/*;q=.2``` (which you are also using) would be the correct way to specify it, and that part does not cause any error.

Examples from the MDN docs linked above:
```
Accept: <MIME_type>/<MIME_subtype>
Accept: <MIME_type>/*
Accept: */*

// Multiple types, weighted with the [quality value](https://developer.mozilla.org/en-US/docs/Glossary/Quality_values) syntax:
Accept: text/html, application/xhtml+xml, application/xml;q=0.9, image/webp, */*;q=0.8
```

**Comment by PatrickG** [comment]
Why not just ignore the invalid ones instead of throwing an error?

**Comment by elikoga** [comment]
> Digging into some old RFCs, it looks like this used to be valid but not anymore (?).
No. See https://datatracker.ietf.org/doc/html/rfc2616#section-2.1

>    *rule
>       The character "*" preceding an element indicates repetition. The
>       full form is "<n>*<m>element" indicating at least <n> and at most
>       <m> occurrences of element. Default values are 0 and infinity so
>       that "*(element)" allows any number, including zero; "1*element"
>       requires at least one; and "1*2element" allows one or two.

**Comment by elikoga** [comment]
> Why not just ignore the invalid ones instead of throwing an error?

Don't really see anything against this, can provide a PR
